lib/datasource/gitlab-tags/index.ts

```diff
@@ -1,136 +1,119 @@
 import { logger } from '../../logger';
-import * as packageCache from '../../util/cache/package';
+import { cache } from '../../util/cache/package/decorator';
 import { GitlabHttp } from '../../util/http/gitlab';
 import { joinUrlParts } from '../../util/url';
+import { Datasource } from '../datasource';
 import type { DigestConfig, GetReleasesConfig, ReleaseResult } from '../types';
 import type { GitlabCommit, GitlabTag } from './types';
 import { defaultRegistryUrl, getDepHost, getSourceUrl } from './util';
 
-export const id = 'gitlab-tags';
-const gitlabApi = new GitlabHttp(id);
+export class GitlabTagsDatasource extends Datasource {
+  static readonly id = 'gitlab-tags';
 
-export const customRegistrySupport = true;
-export const defaultRegistryUrls = [defaultRegistryUrl];
-export const registryStrategy = 'first';
+  protected override http: GitlabHttp;
 
-const cacheNamespace = 'datasource-gitlab';
-
-function getCacheKey(depHost: string, repo: string, type = 'tags'): string {
-  return `${depHost}:${repo}:${type}`;
-}
-
-export async function getReleases({
-  registryUrl,
-  lookupName: repo,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  const depHost = getDepHost(registryUrl);
-
-  const cachedResult = await packageCache.get<ReleaseResult>(
-    cacheNamespace,
-    getCacheKey(depHost, repo)
-  );
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
+  constructor() {
+    super(GitlabTagsDatasource.id);
+    this.http = new GitlabHttp(GitlabTagsDatasource.id);
   }
 
-  const urlEncodedRepo = encodeURIComponent(repo);
-
-  // tag
-  const url = joinUrlParts(
-    depHost,
-    `api/v4/projects`,
-    urlEncodedRepo,
-    `repository/tags?per_page=100`
-  );
-
-  const gitlabTags = (
-    await gitlabApi.getJson<GitlabTag[]>(url, {
-      paginate: true,
-    })
-  ).body;
-
-  const dependency: ReleaseResult = {
-    sourceUrl: getSourceUrl(repo, registryUrl),
-    releases: null,
-  };
-  dependency.releases = gitlabTags.map(({ name, commit }) => ({
-    version: name,
-    gitRef: name,
-    releaseTimestamp: commit?.created_at,
-  }));
-
-  const cacheMinutes = 10;
-  await packageCache.set(
-    cacheNamespace,
-    getCacheKey(depHost, repo),
-    dependency,
-    cacheMinutes
-  );
-  return dependency;
-}
-
-/**
- * gitlab.getDigest
- *
- * This function will simply return the latest commit hash for the configured repository.
- */
-export async function getDigest(
-  { lookupName: repo, registryUrl }: Partial<DigestConfig>,
-  newValue?: string
-): Promise<string | null> {
-  const depHost = getDepHost(registryUrl);
+  override readonly customRegistrySupport = true;
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+  override readonly registryStrategy = 'first';
+
+  @cache({
+    ttlMinutes: 10,
+    namespace: 'datasource-gitlab',
```

Comment:
Is that acceptable to invalidate some amount of cache for the sake of better naming? If so, I need to change this place

Reply:
Ah, never mind, it's just 10 minutes TTL

Reply:
I would omit `ttlMinutes` here anyway. The default is 30 minutes, which is a nice happy-medium for most datasources.